### PR TITLE
Implement typed dictionaries

### PR DESCRIPTION
**Pull request by Repiteo**
- Depends on #1593

A godot-cpp counterpart to PR godotengine/godot#78656

Tests/builds will fail until the above is merged, as this PR is dependant on it.

### CODE REVIEW

**Review by dsnopek — APPROVED** [review]
Thanks! This looks good to me :-)

File: binding_generator.py [hunk 1]
```diff
@@ -2765,6 +2827,12 @@ def correct_typed_array(type_name):
     return type_name
 
 
+def correct_typed_dictionary(type_name):
```
Comment (Zylann) [inline binding_generator.py:2830]: Looks like it's unused?

File: binding_generator.py [hunk 2]
```diff
@@ -1438,6 +1443,32 @@ def generate_engine_classes_bindings(api, output_dir, use_template_get_node):
...
+                                if dict_type_name.endswith("*"):
+                                    dict_type_name = dict_type_name[:-1]
+                                if is_included(dict_type_name, class_name):
+                                    if is_enum(dict_type_name):
+                                        fully_used_classes.add(get_enum_class(dict_type_name))
+                                    elif "default_value" in argument:
+                                        fully_used_classes.add(dict_type_name)
+                                    else:
+                                        used_classes.add(dict_type_name)
+                                dict_type_name = dict_type_names[2]
```
Comment (Zylann) [inline binding_generator.py:1462]: Why index 2? That would be the third entry. I think dictionary only have 2 types?

File: binding_generator.py [hunk 3]
```diff
@@ -1467,6 +1498,32 @@ def generate_engine_classes_bindings(api, output_dir, use_template_get_node):
...
+                            if dict_type_name.endswith("*"):
+                                dict_type_name = dict_type_name[:-1]
+                            if is_included(dict_type_name, class_name):
+                                if is_enum(dict_type_name):
+                                    fully_used_classes.add(get_enum_class(dict_type_name))
+                                elif is_variant(dict_type_name):
+                                    fully_used_classes.add(dict_type_name)
+                                else:
+                                    used_classes.add(dict_type_name)
+                            dict_type_name = dict_type_names[2]
```
Comment (Zylann) [inline binding_generator.py:1517]: Why index 2? That would be the third entry. I think dictionary only have 2 types?

**Pull request merged**